src/python/WMComponent/DBS3Buffer/MySQL/CheckDrainStatus.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import print_function, division
+from WMCore.Database.DBFormatter import DBFormatter
+
+class CheckDrainStatus(DBFormatter):
+    """
+    _CheckDrainStatus_
+    
+    1. checks all the workflows are completed status (within the agent)
+    2. checks all the blocks are closed
+    3. checks all the files are injected to PhEDEx
+    4. checks all the files are uploaded to DBS
+
+    TODO: need to check archived statsus
+    """
+    completedSQL = "SELECT count(*) FROM dbsbuffer_workflow WHERE completed = 0"
+    
+    blockClosedSQL = "SELECT count(*) FROM dbsbuffer_block where status != 'Closed'"
+
+    phedexSQL = "SELECT count(*) FROM dbsbuffer_file WHERE in_phedex = 0"
+    
+    dbsSQL = "SELECT count(*) FROM dbsbuffer_file WHERE status != 'InDBS'"
```

Comment:
We should apply some filters to these two `dbsbuffer_file` queries, since we don't want to get hold by backfill or crappy files that we don't really care. Some examples here:
https://github.com/amaltaro/ProductionTools/blob/master/drainAgent.py#L63